part2/src/array.h

```diff
@@ -106,14 +106,189 @@ class StringArray: public Object {
          * @param obj The object that should be
          * @param i
          */
-        void put(size_t i, String* obj) {}
+        void put(size_t i, String* givenString) {}
 
         /**
          * Returns the first occurrence of the given object in the array using the object's equal() method.
          * If the object is not found, size() is returned.
          * @param obj The object to find in the array
          * @return The first index of the object. size() if not found
          */
-        size_t indexOf(String* obj) {}
+        size_t indexOf(String* givenString) {}
+
+};
+
+/**
+ * An array of ints with a dynamic size
+ */
+class IntArray: public Object {
+    public:
+
+        // NOTE: IntArray can be implementing using ObjectArray for code reuse
+
+        /**
+         * Creates a new object array. All of the elements in the array are initialized to nullptr.
+         * @param size The capacity of the array. Should be > 0, otherwise the program will exit.
+         */
+        IntArray(size_t size) : Object() {}
```

Comment:
We should probably change the constructors to not take capacity since the array is not a fixed size. Might want to do this in this PR so that you dont have to go fix all the tests too

Reply:
Alright sounds good Ill make a new branch and remove that constructor for all.